### PR TITLE
Connect to Astra via TLS w/o using SCB

### LINKED ISSUE #297: CDM tool is not able to connect with ASTRA using the TLS configuration
CDM tool is not able to connect with ASTRA using the TLS configuration.

After updating the `cdm.properties` file as per the documentation then getting below error:

Error detail:
```
24/08/27 11:58:40 INFO ContactPoints: Contact point localhost:29080 resolves to multiple addresses, will use them all ([localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1])
24/08/27 11:58:40 INFO CqlPrepareAsyncProcessor: Adding handler to invalidate cached prepared statements on type changes
24/08/27 11:58:40 INFO ReloadingKeyManagerFactory: Detected updates to KeyStore file /home/enguser/target_details/identity.jks
24/08/27 11:58:40 INFO ReloadingKeyManagerFactory: KeyStore reloading is disabled. If your Cassandra cluster requires client certificates, client application restarts are infrequent, and client certificates have short lifetimes, then your client may fail to re-establish connections to Cassandra hosts. To enable KeyStore reloading, see `advanced.ssl-engine-factory.keystore-reload-interval` in reference.conf.
24/08/27 11:58:40 INFO Clock: Using native clock for microsecond precision
24/08/27 11:58:40 WARN ControlConnection: [s1] Error connecting to Node(endPoint=localhost/127.0.0.1:29080, hostId=null, hashCode=3653990d), trying next node (AnnotatedConnectException: Connection refused: localhost/127.0.0.1:29080)
24/08/27 11:58:40 WARN ControlConnection: [s1] Error connecting to Node(endPoint=localhost/0:0:0:0:0:0:0:1:29080, hostId=null, hashCode=6f45a6b), trying next node (AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:29080)
Exception in thread "main" java.io.IOException: Failed to open native connection to Cassandra at {localhost:29080} :: Could not reach any contact point, make sure you've provided valid addresses (showing first 2 nodes, use getAllErrors() for more): Node(endPoint=localhost/127.0.0.1:29080, hostId=null, hashCode=3653990d): [com.datastax.oss.driver.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:29080], Node(endPoint=localhost/0:0:0:0:0:0:0:1:29080, hostId=null, hashCode=6f45a6b): [com.datastax.oss.driver.shaded.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/0:0:0:0:0:0:0:1:29080]
        at com.datastax.spark.connector.cql.CassandraConnector$.createSession(CassandraConnector.scala:173)
        at com.datastax.spark.connector.cql.CassandraConnector$.$anonfun$sessionCache$1(CassandraConnector.scala:161)
        at com.datastax.spark.connector.cql.RefCountedCache.createNewValueAndKeys(RefCountedCache.scala:32)
        at com.datastax.spark.connector.cql.RefCountedCache.syncAcquire(RefCountedCache.scala:69)
        at com.datastax.spark.connector.cql.RefCountedCache.acquire(RefCountedCache.scala:57)
        at com.datastax.spark.connector.cql.CassandraConnector.openSession(CassandraConnector.scala:81)
        at com.datastax.spark.connector.cql.CassandraConnector.withSessionDo(CassandraConnector.scala:103)
        at com.datastax.cdm.job.Migrate$.$anonfun$execute$1(Migrate.scala:26)
        at com.datastax.cdm.job.Migrate$.$anonfun$execute$1$adapted(Migrate.scala:25)
        at com.datastax.spark.connector.cql.CassandraConnector.$anonfun$withSessionDo$1(CassandraConnector.scala:104)
        at com.datastax.spark.connector.cql.CassandraConnector.closeResourceAfterUse(CassandraConnector.scala:121)
        at com.datastax.spark.connector.cql.CassandraConnector.withSessionDo(CassandraConnector.scala:103)
        at com.datastax.cdm.job.Migrate$.execute(Migrate.scala:25)
        at com.datastax.cdm.job.Migrate$.delayedEndpoint$com$datastax$cdm$job$Migrate$1(Migrate.scala:20)
        at com.datastax.cdm.job.Migrate$delayedInit$body.apply(Migrate.scala:18)
        at scala.Function0.apply$mcV$sp(Function0.scala:39)
        at scala.Function0.apply$mcV$sp$(Function0.scala:39)
        at scala.runtime.AbstractFunction0.apply$mcV$sp(AbstractFunction0.scala:17)
        at scala.App.$anonfun$main$1(App.scala:76)
        at scala.App.$anonfun$main$1$adapted(App.scala:76)
        at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
        at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:926)
        at scala.App.main(App.scala:76)
        at scala.App.main$(App.scala:74)
```

### PR DESCRIPTION
**Pull request by pravinbhat**
**What this PR does**: Implements feature to connect to Astra via TLS w/o using SCB

**Which issue(s) this PR fixes**:
Fixes #297 

**Checklist:**
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/scala/com/datastax/cdm/job/ConnectionFetcher.scala [hunk 1]
```diff
@@ -72,6 +76,17 @@ class ConnectionFetcher(sparkContext: SparkContext, propertyHelper: PropertyHelp
         .set("spark.cassandra.auth.password", connectionDetails.password)
         .set("spark.cassandra.input.consistency.level", consistencyLevel)
         .set("spark.cassandra.connection.config.cloud.path", connectionDetails.scbPath))
+    } else if (connectionDetails.trustStorePath.nonEmpty && connectionDetails.isAstra) {
+      logger.info("Connecting to Astra "+side+" (with truststore) using host metadata at "+connectionDetails.host+":"+connectionDetails.port);
+
+      val scbBundle = generateSCB(connectionDetails.host, connectionDetails.port,
```
Comment (msmygit) [inline src/main/scala/com/datastax/cdm/job/ConnectionFetcher.scala:82]: it's already a bundle.

```suggestion
      val scbFile = generateSCB(connectionDetails.host, connectionDetails.port, 
```

File: src/main/scala/com/datastax/cdm/job/ConnectionFetcher.scala [hunk 2]
```diff
@@ -72,6 +76,17 @@ class ConnectionFetcher(sparkContext: SparkContext, propertyHelper: PropertyHelp
...
+      logger.info("Connecting to Astra "+side+" (with truststore) using host metadata at "+connectionDetails.host+":"+connectionDetails.port);
+
+      val scbBundle = generateSCB(connectionDetails.host, connectionDetails.port, 
+      	connectionDetails.trustStorePassword, connectionDetails.trustStorePath, 
+      	connectionDetails.keyStorePassword, connectionDetails.keyStorePath, side)
+      return CassandraConnector(config
+        .set("spark.cassandra.auth.username", connectionDetails.username)
+        .set("spark.cassandra.auth.password", connectionDetails.password)
+        .set("spark.cassandra.input.consistency.level", consistencyLevel)
+        .set("spark.cassandra.connection.config.cloud.path", "file://" + scbBundle.getAbsolutePath()))
```
Comment (msmygit) [inline src/main/scala/com/datastax/cdm/job/ConnectionFetcher.scala:89]:
```suggestion
        .set("spark.cassandra.connection.config.cloud.path", "file://" + scbFile.getAbsolutePath()))
```

File: src/resources/cdm-detailed.properties [hunk 3]
```diff
@@ -441,6 +441,7 @@ spark.cdm.perfops.ratelimit.target                20000
 #       .path             : Filepath to the Java keystore file
 #       .password         : Password needed to open the keystore
 #    .enabledAlgorithms   : Default is TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA
+#    .isAstra             : Default is false. Set to true if connecting to DataStax Astra without SCB
```
Comment (msmygit) [inline src/resources/cdm-detailed.properties:444]:
```suggestion
#    .isAstra             : Default is false. Set to true if connecting to DataStax Astra DB without SCB
```

**Review by msmygit — APPROVED** [review]

File: src/main/java/com/datastax/cdm/data/DataUtility.java [hunk 4]
```diff
@@ -143,4 +150,68 @@ public static String getMyClassMethodLine(Exception e) {
...
+            out.close();
+
+            return zipfile;
+        } catch (IOException ex) {
+            System.err.println(ex.getMessage());
+        }
+        return null;
+    }
+
+    static class FilePathAndNewName {
```
[the remark this answers is not in the document]
Comment (pravinbhat) [reply src/main/java/com/datastax/cdm/data/DataUtility.java:208]: Its not about separating filename & path. We have to rename the JKS files always as `identity.jks` & `trustStore.jks` for the java-driver code to work. If the enterprise choses to call their JKS files with some other names, we still have to package them with the above static names for the java-driver code to work. This is essentially a java driver bug, but has no impact as of now as the SCB is always generated in that fashion by Astra.

**Review by mfmaher2 — APPROVED** [review]

**Pull request merged**